webrtc/src/main/java/com/nabto/edge/client/webrtc/webrtc.kt

```diff
@@ -0,0 +1,230 @@
+
+    /**
+     * A invalid ICE candidate was received from the other peer
+     */
+    class ICECandidateError() : EdgeWebRTCError()
+
+    /**
+     * The RTC PeerConnection could not be created
+     */
+    class ConnectionInitError() : EdgeWebRTCError()
+}
+
+/**
+ * Video Track representing a Media Track of type Video
+ */
+interface EdgeVideoTrack : EdgeMediaTrack {
+    // TODO: add @throws docs
+    /**
+     * Add a Video View to the track
+     *
+     * @param view [in] The view to add
+     */
+    fun add(view: EdgeVideoView)
+
+    // TODO: add @throws docs
+    /**
+     * remove a Video View to the track
+     *
+     * @param view [in] The view to remove
+     */
+    fun remove(view: EdgeVideoView)
+}
+
+/**
+ * Audio Track representing a Media Track of type Audio
+ */
+interface EdgeAudioTrack : EdgeMediaTrack {
+    // TODO: add @throws docs
+    /**
+     * Enable or disable the Audio track
+     *
+     * @param enabled [in] Boolean determining if the track is enabled
+     */
+    fun setEnabled(enabled: Boolean)
+
+    // TODO: add @throws docs
+    /**
+     * Set the volume of the Audio track
+     *
+     * @param volume [in] The volume to set
+     */
+    fun setVolume(volume: Double)
+}
+
+/**
+ * Callback invoked when a WebRTC connection has been established
+ */
+typealias OnConnectedCallback = () -> Unit
+
+/**
+ * Callback invoked when a WebRTC connection has been closed
+ */
+typealias OnClosedCallback = () -> Unit
+
+/**
+ * Callback invoked when the remote peer has added a Track to the WebRTC connection
+ *
+ * @param EdgeMediaTrack [in] The newly added Track
+ */
+typealias OnTrackCallback = (EdgeMediaTrack) -> Unit
+
+/**
+ * Callback invoked when an error occurs in the WebRTC connection
+ *
+ * @param EdgeWebRTCError [in] The Error that occured
+ */
+typealias OnErrorCallback = (EdgeWebRTCError) -> Unit
+
+
+/**
+ * Main Connection interface used to connect to a device and interact with it.
+ */
+interface EdgeWebrtcConnection {
+
+    /**
+     * Set callback to be invoked when the WebRTC connection is connected
+     *
+     * @param cb The callback to set
+     */
+    fun onConnected(cb: OnConnectedCallback)
+
+    /**
+     * Set callback to be invoked when the WebRTC connection is closed
+     *
+     * @param cb The callback to set
+     */
+    fun onClosed(cb: OnClosedCallback)
+
+    /**
+     * Set callback to be invoked when a new track is available on the WebRTC connection
+     *
+     * @param cb The callback to set
+     */
+    fun onTrack(cb: OnTrackCallback)
+
+    /**
+     * Set callback to be invoked when an error occurs on the WebRTC connection.
+     *
+     * @param cb The callback to set
+     */
+    fun onError(cb: OnErrorCallback)
+
+
+    // TODO: add @throws docs
+    /**
+     * Establish a WebRTC connection to the other peer
+     */
+    fun connect()
+
+    // TODO: add @throws docs
+    /**
+     * Close a connected WebRTC connection.
+     */
+    fun close()
```

Comment:
Is this a AutoCloseable resource? then the interface should extend AutoCloseable? else we should call it something else?